extension-androidauto/build.gradle.kts

```diff
@@ -23,8 +23,8 @@ android {
 }
 
 dependencies {
-  compileOnly(project(":sdk"))
-  testImplementation(project(":sdk"))
+  api(Dependencies.androidAutoMapboxMapSdk)
```

Comment:
Shall we use implementation rather than API?

Reply:
I think we want it to be `api`, specifically because it is surfacing objects from the maps sdk. So that it can be used with or without specifying the maps dependency. 

https://github.com/mapbox/mapbox-maps-android/blob/c8a4da86908ab4decd92a96330f1414bf7b7b6b0/extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarMapSurface.kt#L19

Making it api, means this works for downstream developers
```
implementation("com.mapbox.extension:maps-androidauto:x.y.0)
```

Making it implementation, means downstream developers are required to provide a version of maps in order to use it.
```
implementation("com.mapbox.extension:maps-androidauto:x.y.0")
implementation("com.mapbox.maps:android:10.5.+")
```

Bot approaches are upgradeable
```
implementation("com.mapbox.extension:maps-androidauto:x.y.0")
implementation("com.mapbox.maps:android:10.9.+")
```

Reply:
+1 for having API since underlying maps is not an internal detail and devs most probably need main maps sdk classes at compile time

Reply:
> Making it implementation, means downstream developers are required to provide a version of maps in order to use it.

I even think this is the correct behaviour, as this is just an extension and shouldn't be used independently, user should be very aware of the maps sdk version that they are using under the hood.

Reply:
But in general agree that user should use maps APIs together with the extension, we just need to document the usage well enough.

Reply:
> I even think this is the correct behaviour, as this is just an extension and shouldn't be used independently, user should be very aware of the maps sdk version that they are using under the hood.

However the user could not use the extension _without_ maps, right? And such notation just allows to use shorter notation and skip adding maps dependency explicitly. 
I also assume that most of devs use the default maps version that auto extension depends on, doubt there's much sense in using older version of Maps SDK. And the newer version could be selected, if some specific fix or API is needed.